File: Core API/Team7/Models/Schedule.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Team7.Models
{
    public class Schedule
    {
        public Schedule()
        {
            this.BookingPriceHistory = new HashSet<BookingPriceHistory>();
            this.BookingAttendance = new HashSet<BookingAttendance>();
        }
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int ScheduleID { get; set; }
        public int? VenueID { get; set; }
        public int? BookingTypeID { get; set; }
        public int? LessonID { get; set; }
        public int? EmployeeID { get; set; }

        [Required]
        public System.DateTime StartDateTime { get; set; }
        
        [Required]
        public System.DateTime EndDateTime { get; set; }
        public virtual Venue Venue { get; set; }
        public virtual BookingType BookingType { get; set; }
        public virtual ICollection<BookingAttendance> BookingAttendance { get; set; }
        public virtual ICollection<BookingPriceHistory> BookingPriceHistory { get; set; }
        public virtual Lesson Lesson { get; set; }
        public virtual Employee Employee { get; set; }


    }
}

File: Ionic/src/app/models/schedule.ts
import { BookingType } from "./booking-type";
import { Venue } from "./venue";

export class Schedule {
    scheduleID?: number;
    startDateTime: any;
    endDateTime: any;
    bookingAttendance?: [{
      attended?: boolean;
      bookingAttendanceID?: number;
    }];
    venueID?: number;
    venue?: Venue;
    bookingTypeID?: number;
    bookingType?:BookingType;
    lessonID?: number;
    bookingPriceHistory? :[{
      bookingPriceHistoryID?: number,
      date?: Date;
      amount:number;
    }]
    lesson?: {
      lessonID?: number;
      name?: string;
    };
    employeeID?: number;
    employee?:any;
}
